File: Golang/Palindrome.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

type Palindrome struct {
}

func (*Palindrome) Play() {

	stdScanner := bufio.NewScanner(os.Stdin)
	gameOver := false
	score := 0

	for !gameOver {
		fmt.Println("Enter a word:")

		stdScanner.Scan()

		word := stdScanner.Text()

		result := CheckResult(word)

		if result {
			score++
			fmt.Println("Correct!")
		} else {
			fmt.Println("You've Failed")
			break
		}

		fmt.Println("Your Current Score is ", score)
	}

	fmt.Println("Your Total Score is ", score)

}

func CheckResult(word string) bool {
	wordRunes := []rune(word)
	var reversedWord []rune

	for i := len(word) - 1; i >= 0; i-- {
		reversedWord = append(reversedWord, wordRunes[i])
	}

	if string(reversedWord) == word {
		return true
	} else {
		return false
	}
}

File: C#/Palindrome.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Game
{
    public class Palindrome
    {
        public static void Play()
        {
            var command = "";
            do
            {
                Console.WriteLine("Enter a Word:".ToUpper());
                command = Console.ReadLine();
                var result = CheckResult(command);
                if (result)
                {
                    Console.WriteLine("Succeed");
                }
                else
                {
                    Console.WriteLine("Failed");
                }
            } while (command != "x");
        }

        public static bool CheckResult(string word)
        {
            var wordReverse = "";

            for (int i = word.Length - 1; i >= 0; i--)
            {
                wordReverse += word[i];
            }

            if (word == wordReverse)
            {
                return true;
            }
            else
            {
                return false;
            }
        }
    }
}
